packages/frontend-plugin-api/src/wiring/createExtensionKind.ts

```diff
@@ -0,0 +1,173 @@
+  ): ExtensionKind<TProps, TInputs, TOutput, TConfig> {
+    return new ExtensionKind(options);
+  }
+
+  private constructor(
+    private readonly options: ExtensionKindOptions<
+      TProps,
+      TInputs,
+      TOutput,
+      TConfig
+    >,
+  ) {}
+
+  public new(options: {
+    namespace?: string;
+    name?: string;
+    attachTo?: { id: string; input: string };
+    disabled?: boolean;
+    inputs?: TInputs;
+    output?: TOutput;
+    configSchema?: PortableSchema<TConfig>;
+    props: TProps;
+    factory?(
+      options: {
+        node: AppNode;
+        config: TConfig;
+        inputs: Expand<ResolvedExtensionInputs<TInputs>>;
+        orignalFactory(
+          options?: {
+            node?: AppNode;
+            config?: TConfig;
+            inputs?: Expand<ResolvedExtensionInputs<TInputs>>;
+          },
+          props?: TProps,
+        ): Expand<ExtensionDataValues<TOutput>>;
+      },
+      props: TProps,
+    ): Expand<ExtensionDataValues<TOutput>>;
+  }) {
+    return createExtension({
+      kind: this.options.kind,
+      namespace: options.namespace ?? this.options.namespace,
+      name: options.name ?? this.options.name,
+      attachTo: options.attachTo ?? this.options.attachTo,
+      disabled: options.disabled ?? this.options.disabled,
+      inputs: options.inputs ?? this.options.inputs,
+      output: options.output ?? this.options.output,
+      configSchema: options.configSchema ?? this.options.configSchema, // TODO: some config merging or smth
+      factory: ({ node, config, inputs }) => {
+        if (options.factory) {
+          return options.factory(
+            {
+              node,
+              config,
+              inputs,
+              orignalFactory: (
+                innerOptions?: {
+                  node?: AppNode;
+                  config?: TConfig;
+                  inputs?: Expand<ResolvedExtensionInputs<TInputs>>;
+                },
+                innerProps?: TProps,
+              ) =>
+                this.options.factory(
+                  {
+                    node: innerOptions?.node ?? node,
+                    config: innerOptions?.config ?? config,
+                    inputs: innerOptions?.inputs ?? inputs,
+                  },
+                  innerProps ?? options.props,
+                ),
+            },
+            options.props,
+          );
+        }
+
+        return this.options.factory(
+          {
+            node,
+            config,
+            inputs,
+          },
+          options.props,
+        );
+      },
+    });
+  }
+}
+
+/**
+ * A simpler replacement for wrapping up `createExtension` inside a kind or type. This allows for a cleaner API for creating
+ * types and instances of those types.
+ *
+ * @public
+ */
+export function createExtensionKind<
+  TProps,
+  TInputs extends AnyExtensionInputMap,
+  TOutput extends AnyExtensionDataMap,
+  TConfig,
+>(options: ExtensionKindOptions<TProps, TInputs, TOutput, TConfig>) {
```

Comment:
if you make the ExtensionKind an interface, maybe put it as an explicit return type here to facilitate the types a bit

Reply:
will pick this back up later.